snagflash: dfu: add `--dfu-keep`, `--dfu-detach` and `--dfu-reset` to manage DFU mode

AdrianDC opened the pull request:
## Purpose

For specific use cases, sequential commands to flash a board through DFU,
and to match standard DFU documentations where the detach event is called manually,
add these new arguments to keep and detach the DFU mode manually.

## Example

STM32 booting in DFU mode, without an UART debugging (unwished or unavailable),
flashing multiple altsettings separately or retrying upon error without using `dfu 0` every time in UART.

rgantois commented on src/snagflash/dfu.py:
```diff
@@ -32,22 +32,24 @@ def dfu_cli(args):
-		logger.debug(f"DFU config altsetting:{altsetting} size:0x{size:x} path:{path}")
+	altsetting = 0
+	if args.dfu_config:
+		for dfu_config in args.dfu_config:
```
Could you please move the body of this `for` loop into a separate function? The indentation is getting a little too deep here.

AdrianDC replied:
Fully refactored and reused for `--dfu-reset`

rgantois reviewed (changes requested):
Thanks for the PR! I think these will be useful features to have for DFU mode.

AdrianDC commented:
Updated and added `--dfu-reset` support to reset the board from DFU mode.

Tested on an STM32MP157CAC to automatically :
* Recover (optional, if in recovery mode) or prepare DFU from eMMC U-Boot
* Flash TSV
* Flash all partitions over altsettings
* Reboot to system automatically

rgantois commented:
Nice!

AdrianDC commented:
:+1: 

Any expected date for a 1.2 release ? Thanks

rgantois commented:
I'm planning to release it sometime this month.
